sigar_unix.go

```diff
@@ -22,3 +27,52 @@ func (self *FileSystemUsage) Get(path string) error {
 
 	return nil
 }
+
+func (r *Rusage) Get(who int) error {
+	ru, err := getResourceUsage(who)
+	if err != nil {
+		return err
+	}
+
+	uTime, err := convertRtimeToDur(ru.Utime)
+	if err != nil {
+		return err
+	}
+	sTime, err := convertRtimeToDur(ru.Stime)
+	if err != nil {
+		return err
+	}
+
+	r.Utime = uTime
+	r.Stime = sTime
+	r.Maxrss = int64(ru.Maxrss)
+	r.Ixrss = int64(ru.Ixrss)
+	r.Idrss = int64(ru.Idrss)
+	r.Isrss = int64(ru.Isrss)
+	r.Minflt = int64(ru.Minflt)
+	r.Majflt = int64(ru.Majflt)
+	r.Nswap = int64(ru.Nswap)
+	r.Inblock = int64(ru.Inblock)
+	r.Oublock = int64(ru.Oublock)
+	r.Msgsnd = int64(ru.Msgsnd)
+	r.Msgrcv = int64(ru.Msgrcv)
+	r.Nsignals = int64(ru.Nsignals)
+	r.Nvcsw = int64(ru.Nvcsw)
+	r.Nivcsw = int64(ru.Nivcsw)
+
+	return nil
+}
+
+func getResourceUsage(who int) (unix.Rusage, error) {
+	r := unix.Rusage{}
+	err := unix.Getrusage(who, &r)
+
+	return r, err
+}
+
+func convertRtimeToDur(t unix.Timeval) (time.Duration, error) {
+	p := fmt.Sprintf("%ds%dus", t.Sec, t.Usec)
```

Comment:
This is quite hacky. How about doing the math to convert the value to nanaseconds instead?

Reply:
Or better yet, use the `Nano()` method of Timeval. https://godoc.org/golang.org/x/sys/unix#Timeval